flag/flag.go

```diff
@@ -356,25 +393,21 @@ func (s *Set) Value(t *dials.Type) (reflect.Value, error) {
 	var setErr error
 	val := reflect.New(t.Type())
 	s.Flags.Visit(func(f *flag.Flag) {
-		fieldIdxs, ok := s.fieldPaths[f.Name]
+		fieldName, ok := s.flagFieldName[f.Name]
 		if !ok {
 			return
 		}
-		// Iteratively initialize unset values
-		for i := 1; i < len(fieldIdxs); i++ {
-			interimField := val.Elem().FieldByIndex(fieldIdxs[:i])
-			if !interimField.IsNil() {
-				// If this field isn't nil, continue, we don't
-				// want to overwrite a populated field with an
-				// empty one. (since that would prevent us from
-				// seeing other fields in the same (or
-				// parallel) sub-structs.
-				continue
-			}
-			interimPtr := reflect.New(interimField.Type().Elem())
-			interimField.Set(interimPtr)
+
+		ffield := s.trnslVal.FieldByName(fieldName)
+		if !ffield.IsNil() {
+			// If this field isn't nil, continue, we don't
+			// want to overwrite a populated field with an
+			// empty one. (since that would prevent us from
+			// seeing other fields in the same (or
+			// parallel) sub-structs.
+			return
```

Comment:
If this happens we have a bug, since there's now a 1:1 mapping between flags and fields.
Since both field-names and flag-names should be unique, this should be impossible.
Let's panic here, (at least until someone complains about their code `panic`ing in production :) )

Reply:
Indeed, added a panic